File: app/src/main/kotlin/com/apex/trader/presentation/screen/about/AboutScreen.kt
package com.apex.trader.presentation.screen.about

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Email
import androidx.compose.material.icons.filled.ShowChart
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.apex.trader.presentation.component.ApexCard
import com.apex.trader.presentation.theme.ApexBull
import com.apex.trader.presentation.theme.ApexHighlight
import com.apex.trader.presentation.theme.ApexPrimary
import com.apex.trader.presentation.theme.ApexPrimaryDim
import com.apex.trader.presentation.theme.ApexTextMuted

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AboutScreen(onBack: () -> Unit) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("About") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background
                )
            )
        }
    ) { padding ->
        LazyColumn(
            modifier = Modifier.fillMaxSize().padding(padding),
            contentPadding = PaddingValues(16.dp),
            verticalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            item { HeroBlock() }
            item { DeveloperCard() }
            item { TechCard() }
            item { DisclaimerCard() }
            item { Spacer(Modifier.height(24.dp)) }
        }
    }
}

@Composable
private fun HeroBlock() {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .background(
                brush = Brush.verticalGradient(listOf(ApexPrimary, ApexPrimaryDim)),
                shape = RoundedCornerShape(20.dp)
            )
            .padding(24.dp),
        contentAlignment = Alignment.Center
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Box(
                modifier = Modifier
                    .size(72.dp)
                    .background(MaterialTheme.colorScheme.background.copy(alpha = 0.18f), RoundedCornerShape(20.dp)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    Icons.Default.ShowChart,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onPrimary,
                    modifier = Modifier.size(40.dp)
                )
            }
            Spacer(Modifier.height(12.dp))
            Text(
                "Apex Trader",
                color = MaterialTheme.colorScheme.onPrimary,
                style = MaterialTheme.typography.headlineMedium,
                fontWeight = FontWeight.Bold
            )
            Text(
                "Binance Futures Confluence Scanner",
                color = MaterialTheme.colorScheme.onPrimary.copy(alpha = 0.85f),
                style = MaterialTheme.typography.bodyMedium,
                textAlign = TextAlign.Center
            )
            Spacer(Modifier.height(6.dp))
            Text(
                "v1.0.0",
                color = MaterialTheme.colorScheme.onPrimary.copy(alpha = 0.7f),
                style = MaterialTheme.typography.labelMedium
            )
        }
    }
}

@Composable
private fun DeveloperCard() {
    ApexCard {
        Column {
            Text(
                "DEVELOPED BY",
                color = ApexTextMuted,
                style = MaterialTheme.typography.labelSmall
            )
            Spacer(Modifier.height(4.dp))
            Text(
                "Muhannnad Waleed Hassoun",
                style = MaterialTheme.typography.titleLarge,
                fontWeight = FontWeight.SemiBold
            )
            Spacer(Modifier.height(10.dp))
            androidx.compose.foundation.layout.Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(
                    Icons.Default.Email,
                    contentDescription = null,
                    tint = ApexHighlight,
                    modifier = Modifier.size(18.dp)
                )
                Spacer(Modifier.size(8.dp))
                Text(
                    "bugmuha@gmail.com",
                    color = ApexHighlight,
                    style = MaterialTheme.typography.bodyLarge
                )
            }
        }
    }
}

@Composable
private fun TechCard() {
    ApexCard {
        Column {
            Text("STRATEGY", color = ApexTextMuted, style = MaterialTheme.typography.labelSmall)
            Spacer(Modifier.height(4.dp))
            Text(
                "Apex Confluence Strategy (ACS)",
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.SemiBold
            )
            Spacer(Modifier.height(6.dp))
            Text(
                "Multi-timeframe confluence: HTF (4H) + MTF (1H) trend bias with " +
                    "an 11-component LTF (15m) entry trigger. ATR-based dynamic " +
                    "stop-loss and 1.5R / 2.5R / 4.0R take-profits. Only signals " +
                    "scoring ≥ 70% confidence are surfaced.",
                color = ApexTextMuted,
                style = MaterialTheme.typography.bodyMedium
            )
            Spacer(Modifier.height(12.dp))
            Text("BUILT WITH", color = ApexTextMuted, style = MaterialTheme.typography.labelSmall)
            Spacer(Modifier.height(4.dp))
            Text(
                "Kotlin 2.0  ·  Jetpack Compose  ·  Hilt  ·  Coroutines  ·  Retrofit  ·  WorkManager",
                color = ApexBull,
                style = MaterialTheme.typography.bodyMedium
            )
        }
    }
}

@Composable
private fun DisclaimerCard() {
    ApexCard {
        Column {
            Text("DISCLAIMER", color = ApexTextMuted, style = MaterialTheme.typography.labelSmall)
            Spacer(Modifier.height(4.dp))
            Text(
                "This app sends real orders to the Binance Futures API. Derivatives " +
                    "trading carries substantial risk and you can lose more than your " +
                    "initial deposit. Past performance does not guarantee future results. " +
                    "Test on Binance Testnet first. Use at your own risk.",
                color = ApexTextMuted,
                style = MaterialTheme.typography.bodyMedium
            )
        }
    }
}

File: lib/features/about/about_screen.dart
import 'package:flutter/material.dart';

import '../../core/theme.dart';
import '../../widgets/common.dart';

class AboutScreen extends StatelessWidget {
  const AboutScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('About')),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          _hero(context),
          const SizedBox(height: 14),
          _developer(context),
          const SizedBox(height: 14),
          _tech(context),
          const SizedBox(height: 14),
          _disclaimer(context),
          const SizedBox(height: 24),
        ],
      ),
    );
  }

  Widget _hero(BuildContext context) {
    return Container(
      padding: const EdgeInsets.all(24),
      decoration: BoxDecoration(
        gradient: const LinearGradient(
          begin: Alignment.topCenter,
          end: Alignment.bottomCenter,
          colors: [ApexColors.primary, ApexColors.primaryDim],
        ),
        borderRadius: BorderRadius.circular(20),
      ),
      child: Column(
        children: [
          Container(
            width: 72,
            height: 72,
            decoration: BoxDecoration(
              color: ApexColors.background.withValues(alpha: 0.18),
              borderRadius: BorderRadius.circular(20),
            ),
            child: const Icon(Icons.show_chart, color: Colors.white, size: 40),
          ),
          const SizedBox(height: 12),
          Text(
            'Apex Trader',
            style: Theme.of(context).textTheme.headlineMedium?.copyWith(
                  color: Colors.white,
                  fontWeight: FontWeight.w800,
                ),
          ),
          const Text(
            'Binance Futures Confluence Scanner',
            textAlign: TextAlign.center,
            style: TextStyle(color: Colors.white70),
          ),
          const SizedBox(height: 6),
          const Text('v1.0.0',
              style: TextStyle(color: Colors.white60, fontSize: 12, fontFamily: 'monospace')),
        ],
      ),
    );
  }

  Widget _developer(BuildContext context) => ApexCard(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SectionLabel('DEVELOPED BY'),
            const SizedBox(height: 4),
            Text('Muhannnad Waleed Hassoun',
                style: Theme.of(context).textTheme.titleLarge?.copyWith(fontWeight: FontWeight.w600)),
            const SizedBox(height: 10),
            Row(
              children: const [
                Icon(Icons.email_outlined, color: ApexColors.highlight, size: 18),
                SizedBox(width: 8),
                Text('bugmuha@gmail.com',
                    style: TextStyle(color: ApexColors.highlight, fontSize: 15)),
              ],
            ),
          ],
        ),
      );

  Widget _tech(BuildContext context) => ApexCard(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SectionLabel('STRATEGY'),
            const SizedBox(height: 4),
            Text('Apex Confluence Strategy (ACS)',
                style: Theme.of(context).textTheme.titleMedium?.copyWith(fontWeight: FontWeight.w600)),
            const SizedBox(height: 6),
            const Text(
              'Multi-timeframe confluence: HTF (4H) + MTF (1H) trend bias with an '
              '11-component LTF (15m) entry trigger. ATR-based dynamic stop-loss '
              'and 1.5R / 2.5R / 4.0R take-profits. Only signals scoring ≥ 70% '
              'confidence are surfaced.',
              style: TextStyle(color: ApexColors.textMuted),
            ),
            const SizedBox(height: 12),
            const SectionLabel('BUILT WITH'),
            const SizedBox(height: 4),
            const Text(
              'Flutter  ·  Dart 3  ·  Riverpod  ·  Dio  ·  fl_chart',
              style: TextStyle(color: ApexColors.bull),
            ),
          ],
        ),
      );

  Widget _disclaimer(BuildContext context) => const ApexCard(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            SectionLabel('DISCLAIMER'),
            SizedBox(height: 4),
            Text(
              'This app sends real orders to the Binance Futures API. Derivatives '
              'trading carries substantial risk and you can lose more than your '
              'initial deposit. Past performance does not guarantee future results. '
              'Test on Binance Testnet first. Use at your own risk.',
              style: TextStyle(color: ApexColors.textMuted),
            ),
          ],
        ),
      );
}
